app.py

```diff
@@ -228,23 +216,69 @@ def add_plant_info(scientific_name_raw: str, common_name_raw: str, region_raw: s
     """
     scientific_name = escape(scientific_name_raw)
     common_name = escape(common_name_raw)
-    region = escape(region_raw)
+    region_name = escape(region_raw)
 
     conn = psycopg2.connect(app.config['CONNECTION_STRING'])
     with conn:
         with conn.cursor() as cursor:
+
+            #Check if plant already exists
+            cursor.execute('SELECT * '
+                            'FROM Plant '
+                            'Where scientificName = %s AND commonName = %s;', 
+                            (scientific_name, common_name)) 
+            plant_row = cursor.fetchone()
+            
+            if plant_row is not None:
+                return {
+                    'error': 404,
+                    'message': 'Plant record already exists!'
+                }
+            
+            # Check if region already exists
+            #  otherwise insert a new region first
+            helper_keys = ['abbr', 'name']
+            cursor.execute('SELECT * ' 
+                            'FROM Region R '
+                            'WHERE R.name = %s;', (region_name,))
+            region_row = cursor.fetchone()
+            region_info = dict(zip(helper_keys, region_row))
+
+            # Add new region specified by the user
+            if region_row is None:
+                if len(region_name) < 5:
+                    return {
+                        'error': 404,
+                        'message': 'Invalid region name'
+                    }
+                
+                region_abbr = region_name[0:3]
+                try:
+                    cursor.execute('INSERT INTO Region (abbr, name) '
+                                    'VALUES (%s, %s);', (region_abbr, region_name))
+                except (Exception, psycopg2.DatabaseError) as error:
+                    return {
+                        'message': error
+                    }
+            else:
+                # Get the abbr to insert new plant record
+                region_abbr = str(region_info.pop('abbr'))
+
+            # Insert new plant record
             try:
                 cursor.execute('INSERT INTO PLANT (scientificName, commonName, region)'
-                                'VALUE (%s, %s, %s);',
-                                (scientific_name, common_name, region))
+                            'VALUES (%s, %s, %s);',
+                            (scientific_name, common_name, region_abbr))
             except (Exception, psycopg2.DatabaseError) as error:
                 return {
                     'message': error
-                }, 500
+                }
+                    
     conn.close()
     return {
-        'ok': 'true'
-    }, 201
+        'ok': 'true',
+        'message': 'inserted successfully'
+    }
 
 @app.route('/plants/<symptom_name_raw>/plantIllness', methods=['GET'])
```

Comment:
See above comment on illnesses, we should either break this down to `/symptoms/<symptom_name_raw>/illnesses` or extend it to `/plant/<scientific_name_raw>/symptoms/<symptom_name_raw>/illnesses` for clarity.